### PR TITLE
#373 | Remove awaitilty dependency in hermes-frontend

### PR DESCRIPTION
**Pull request by sbolimowski**

### CODE REVIEW

File: hermes-common/src/main/java/pl/allegro/tech/hermes/common/config/Configs.java [hunk 1]
```diff
@@ -73,6 +73,7 @@
     FRONTEND_BACKLOG_SIZE("frontend.backlog.size", 10000),
     FRONTEND_IO_THREADS_COUNT("frontend.io.threads.count", Runtime.getRuntime().availableProcessors() * 2),
     FRONTEND_WORKER_THREADS_COUNT("frontend.worker.threads.count", 200),
+    FRONTEND_COMMON_THREAD_POOL_SIZE("frontend.common.thread.pool.size", 8),
```
Comment (adamdubiel) [inline hermes-common/src/main/java/pl/allegro/tech/hermes/common/config/Configs.java]: `FRONTEND_TOPIC_LOADER_THREAD_POOL` ?

File: hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/buffer/BackupMessagesLoader.java [hunk 2]
```diff
@@ -68,6 +74,7 @@ public BackupMessagesLoader(BrokerMessageProducer brokerMessageProducer,
         this.messageMaxAgeHours = config.getIntProperty(MESSAGES_LOCAL_STORAGE_MAX_AGE_HOURS);
         this.resendSleep = config.getIntProperty(KAFKA_PRODUCER_ACK_TIMEOUT) + secondsToWaitForTopicsCache * 1000;
         this.maxResendRetries = config.getIntProperty(MESSAGES_LOCAL_STORAGE_MAX_RESEND_RETRIES);
+        this.executorService = executorService;
     }
 
     public void loadMessages(MessageRepository messageRepository) {
```
Comment (adamdubiel) [inline hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/buffer/BackupMessagesLoader.java]: we could create the thread pool and close it in finally in this method, not to allocate resources for the lifetime of Frontend instance

File: hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/buffer/BackupMessagesLoader.java [hunk 3]
```diff
@@ -80,7 +87,7 @@ public void loadMessages(MessageRepository messageRepository) {
 
         do {
             if (retry > 0) {
-                List<Pair<Message, Topic>> retryMessages = new ArrayList<>(toResend.getAndSet(new ConcurrentLinkedQueue<>()));
+                List<Pair<Message, Topic>> retryMessages = Lists.newArrayList(toResend.getAndSet(new ConcurrentLinkedQueue<>()));
```
Comment (chemicL) [inline hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/buffer/BackupMessagesLoader.java]: Not sure why the change

File: hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/buffer/BackupMessagesLoader.java [hunk 4]
```diff
@@ -96,19 +103,26 @@ public void loadMessages(MessageRepository messageRepository) {
 
     private void sendMessages(List<BackupMessage> messages) {
         logger.info("Sending {} messages from backup storage.", messages.size());
+        ExecutorService executor = Executors.newFixedThreadPool(16);
```
Comment (chemicL) [inline hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/buffer/BackupMessagesLoader.java]: Why 16?

File: hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/buffer/BackupMessagesLoader.java (hunk 3 above)
Comment (sbolimowski) [inline hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/buffer/BackupMessagesLoader.java]: To remove compiler warning

File: hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/buffer/BackupMessagesLoader.java (hunk 4 above)
Comment (sbolimowski) [inline hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/buffer/BackupMessagesLoader.java]: Just a size for beginning, can be tuned in the future, if application start will take too long.

**Comment by chemicL** [comment]
👍

File: hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/buffer/BackupMessagesLoader.java [hunk 5]
```diff
@@ -96,19 +103,26 @@ public void loadMessages(MessageRepository messageRepository) {
...
-            Optional<Topic> topic = loadTopic(fromQualifiedName(backupMessage.getQualifiedTopicName()));
+            Optional<Topic> topic = loadTopic(fromQualifiedName(backupMessage.getQualifiedTopicName()), executor);
             if (sendMessageIfNeeded(message, topic, "sending")) {
                 sentCounter++;
             } else {
                 discardedCounter++;
             }
         }
 
+        executor.shutdownNow();
```
Comment (adamdubiel) [inline hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/buffer/BackupMessagesLoader.java]: shouldn't we free those resources in some finally block just to be on the safe side?

**Comment by adamdubiel** [comment]
👍
